tools/twix/src/repository_parameters.rs

```diff
@@ -90,11 +105,20 @@ fn nest_value_at_path(path: &str, value: &Value) -> Value {
         })
 }
 
+fn get_diff_against_stored_value(stored_value: &Value, incoming_sparse_value: &Value) -> Value {
+    let mut diff = stored_value.clone();
+    merge_json(&mut diff, incoming_sparse_value);
+    prune_equal_branches(&mut diff, stored_value);
+    diff
+}
```

Comment:
Why not take `stored_value` as owned value and make it `mut` inside this function. No need to clone. The borrowed value is not used anywhere else so you should be able to move it in this function.

```suggestion
fn get_diff_against_stored_value(mut stored_value: Value, incoming_sparse_value: &Value) -> Value {
    merge_json(&mut stored_value, incoming_sparse_value);
    prune_equal_branches(&mut stored_value, stored_value);
    stored_value
}
```

Reply:
Unfortunately, this wouldn't work as I need the original stored value to perform the pruning `prune_equal_branches(&mut stored_value, stored_value);`.

Reply:
Ah, I see. Mist.